server/queues/processors/WebsocketsProcessor.ts

```diff
@@ -466,153 +480,172 @@ export default class WebsocketsProcessor {
-        // result of the group deletion since there are cascading deletes, we
-        // approximate this by looking for the recently deleted items in the
-        // GroupUser and CollectionGroup tables
-        const groupUsers = await GroupUser.findAll({
-          paranoid: false,
+        const groupMemberships = await GroupMembership.findAll({
           where: {
             groupId: event.modelId,
-            deletedAt: {
-              [Op.gt]: subHours(new Date(), 1),
-            },
           },
         });
-        const groupMemberships = await GroupMembership.scope(
-          "withCollection"
-        ).findAll({
-          paranoid: false,
-          where: {
-            groupId: event.modelId,
-            deletedAt: {
-              [Op.gt]: subHours(new Date(), 1),
+
+        await GroupUser.findAllInBatches<GroupUser>(
+          {
+            where: {
+              groupId: event.modelId,
             },
+            include: [
+              {
+                association: "user",
+                required: true,
+              },
+            ],
+            limit: 100,
           },
-        });
-
-        for (const groupMembership of groupMemberships) {
-          const membershipUserIds = groupMembership.collectionId
-            ? await Collection.membershipUserIds(groupMembership.collectionId)
-            : [];
-
-          for (const groupUser of groupUsers) {
-            if (membershipUserIds.includes(groupUser.userId)) {
-              // the user still has access through some means...
-              // treat this like an add, so that the client re-syncs policies
-              socketio
-                .to(`user-${groupUser.userId}`)
-                .emit("collections.add_user", {
-                  event: event.name,
-                  userId: groupUser.userId,
-                  collectionId: groupMembership.collectionId,
-                });
-            } else {
-              // let everyone with access to the collection know a user was removed
-              socketio
-                .to(`collection-${groupMembership.collectionId}`)
-                .emit("collections.remove_user", {
-                  event: event.name,
-                  userId: groupUser.userId,
-                  collectionId: groupMembership.collectionId,
-                });
-              // tell any user clients to disconnect from the websocket channel for the collection
-              socketio.to(`user-${groupUser.userId}`).emit("leave", {
-                event: event.name,
-                collectionId: groupMembership.collectionId,
-              });
+          async (groupUsers) => {
+            for (const groupMembership of groupMemberships) {
+              if (!groupMembership.collectionId) {
+                continue;
+              }
+              const payload = presentGroupMembership(groupMembership);
+
+              for (const groupUser of groupUsers) {
+                socketio
```

Comment:
Unfortunately there is not enough information on the client to cleanup from a `groups.delete` event alone, so we send `collections.remove_group` for each individual collection (and soon document) membership instead.